handlers/handlers.go

```diff
@@ -57,6 +65,60 @@ func wrapGetWithErrors(ef func(error), hf handlerFunc) http.Handler {
 	})
 }
 
+func hmacAuth(hf handlerFunc, secretKey string, serviceId string) handlerFunc {
```

Comment:
Please add a docstring (see other functions in this file for example), something like

```
hmacAuth wraps incoming requests to validate that the request includes the appropriate 
signature, if the server requires signed requests...
```